frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -1,24 +1,44 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues } from 'formik';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import { safeDump } from 'js-yaml';
 import { ItemSelectorField } from '@console/shared';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
 import { NormalizedEventSources } from '../import-types';
-import { getEventSourceData } from '../../../utils/create-eventsources-utils';
+import {
+  getEventSourceData,
+  getEventSourceYaml,
+  EditorType,
+} from '../../../utils/create-eventsources-utils';
 
 interface EventSourcesSelectorProps {
   eventSourceList: NormalizedEventSources;
 }
 
 const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSourceList }) => {
   const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
+  const [applicationField] = useField('application.name');
+  const [namespaceField] = useField('project.name');
   const handleItemChange = React.useCallback(
     (item: string) => {
       const name = `data.${item.toLowerCase()}`;
       setFieldValue(name, getEventSourceData(item.toLowerCase()));
       setFieldTouched(name, true);
+      setFieldValue('editorType', EditorType.Form);
+      setFieldTouched('editorType', true);
+      if (item.toLowerCase() === 'camelsource') {
```

Comment:
This seems setting to be handling specific logic for a specific type here in this genetic component.

Reply:
changed it to check and load form or yaml editor values for any event source selected